// skywalking-backend-js/lib/proto/management/Management_pb.js
"use strict";
// source: management/Management.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();
var common_Common_pb = require('../common/Common_pb.js');
goog.object.extend(proto, common_Common_pb);
goog.exportSymbol('proto.InstancePingPkg', null, global);
goog.exportSymbol('proto.InstanceProperties', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.InstanceProperties = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, proto.InstanceProperties.repeatedFields_, null);
};
goog.inherits(proto.InstanceProperties, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.InstanceProperties.displayName = 'proto.InstanceProperties';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.InstancePingPkg = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.InstancePingPkg, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.InstancePingPkg.displayName = 'proto.InstancePingPkg';
}
/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.InstanceProperties.repeatedFields_ = [3];
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.InstanceProperties.prototype.toObject = function (opt_includeInstance) {
        return proto.InstanceProperties.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.InstanceProperties} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.InstanceProperties.toObject = function (includeInstance, msg) {
        var f, obj = {
            service: jspb.Message.getFieldWithDefault(msg, 1, ""),
            serviceinstance: jspb.Message.getFieldWithDefault(msg, 2, ""),
            propertiesList: jspb.Message.toObjectList(msg.getPropertiesList(), common_Common_pb.KeyStringValuePair.toObject, includeInstance)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.InstanceProperties}
 */
proto.InstanceProperties.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.InstanceProperties;
    return proto.InstanceProperties.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.InstanceProperties} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.InstanceProperties}
 */
proto.InstanceProperties.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setService(value);
                break;
            case 2:
                var value = /** @type {string} */ (reader.readString());
                msg.setServiceinstance(value);
                break;
            case 3:
                var value = new common_Common_pb.KeyStringValuePair;
                reader.readMessage(value, common_Common_pb.KeyStringValuePair.deserializeBinaryFromReader);
                msg.addProperties(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.InstanceProperties.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.InstanceProperties.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.InstanceProperties} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.InstanceProperties.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getService();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getServiceinstance();
    if (f.length > 0) {
        writer.writeString(2, f);
    }
    f = message.getPropertiesList();
    if (f.length > 0) {
        writer.writeRepeatedMessage(3, f, common_Common_pb.KeyStringValuePair.serializeBinaryToWriter);
    }
};
/**
 * optional string service = 1;
 * @return {string}
 */
proto.InstanceProperties.prototype.getService = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.InstanceProperties} returns this
 */
proto.InstanceProperties.prototype.setService = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional string serviceInstance = 2;
 * @return {string}
 */
proto.InstanceProperties.prototype.getServiceinstance = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};
/**
 * @param {string} value
 * @return {!proto.InstanceProperties} returns this
 */
proto.InstanceProperties.prototype.setServiceinstance = function (value) {
    return jspb.Message.setProto3StringField(this, 2, value);
};
/**
 * repeated KeyStringValuePair properties = 3;
 * @return {!Array<!proto.KeyStringValuePair>}
 */
proto.InstanceProperties.prototype.getPropertiesList = function () {
    return /** @type{!Array<!proto.KeyStringValuePair>} */ (jspb.Message.getRepeatedWrapperField(this, common_Common_pb.KeyStringValuePair, 3));
};
/**
 * @param {!Array<!proto.KeyStringValuePair>} value
 * @return {!proto.InstanceProperties} returns this
*/
proto.InstanceProperties.prototype.setPropertiesList = function (value) {
    return jspb.Message.setRepeatedWrapperField(this, 3, value);
};
/**
 * @param {!proto.KeyStringValuePair=} opt_value
 * @param {number=} opt_index
 * @return {!proto.KeyStringValuePair}
 */
proto.InstanceProperties.prototype.addProperties = function (opt_value, opt_index) {
    return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.KeyStringValuePair, opt_index);
};
/**
 * Clears the list making it empty but non-null.
 * @return {!proto.InstanceProperties} returns this
 */
proto.InstanceProperties.prototype.clearPropertiesList = function () {
    return this.setPropertiesList([]);
};
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.InstancePingPkg.prototype.toObject = function (opt_includeInstance) {
        return proto.InstancePingPkg.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.InstancePingPkg} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.InstancePingPkg.toObject = function (includeInstance, msg) {
        var f, obj = {
            service: jspb.Message.getFieldWithDefault(msg, 1, ""),
            serviceinstance: jspb.Message.getFieldWithDefault(msg, 2, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.InstancePingPkg}
 */
proto.InstancePingPkg.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.InstancePingPkg;
    return proto.InstancePingPkg.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.InstancePingPkg} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.InstancePingPkg}
 */
proto.InstancePingPkg.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {string} */ (reader.readString());
                msg.setService(value);
                break;
            case 2:
                var value = /** @type {string} */ (reader.readString());
                msg.setServiceinstance(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.InstancePingPkg.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.InstancePingPkg.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.InstancePingPkg} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.InstancePingPkg.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getService();
    if (f.length > 0) {
        writer.writeString(1, f);
    }
    f = message.getServiceinstance();
    if (f.length > 0) {
        writer.writeString(2, f);
    }
};
/**
 * optional string service = 1;
 * @return {string}
 */
proto.InstancePingPkg.prototype.getService = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};
/**
 * @param {string} value
 * @return {!proto.InstancePingPkg} returns this
 */
proto.InstancePingPkg.prototype.setService = function (value) {
    return jspb.Message.setProto3StringField(this, 1, value);
};
/**
 * optional string serviceInstance = 2;
 * @return {string}
 */
proto.InstancePingPkg.prototype.getServiceinstance = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};
/**
 * @param {string} value
 * @return {!proto.InstancePingPkg} returns this
 */
proto.InstancePingPkg.prototype.setServiceinstance = function (value) {
    return jspb.Message.setProto3StringField(this, 2, value);
};
goog.object.extend(exports, proto);
