// skywalking-backend-js/lib/proto/management/Management_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!
// Original file comments:
//
// Licensed to the Apache Software Foundation (ASF) under one or more
// contributor license agreements.  See the NOTICE file distributed with
// this work for additional information regarding copyright ownership.
// The ASF licenses this file to You under the Apache License, Version 2.0
// (the "License"); you may not use this file except in compliance with
// the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
//
'use strict';
var grpc = require('grpc');
var management_Management_pb = require('../management/Management_pb.js');
var common_Common_pb = require('../common/Common_pb.js');
function serialize_Commands(arg) {
    if (!(arg instanceof common_Common_pb.Commands)) {
        throw new Error('Expected argument of type Commands');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_Commands(buffer_arg) {
    return common_Common_pb.Commands.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_InstancePingPkg(arg) {
    if (!(arg instanceof management_Management_pb.InstancePingPkg)) {
        throw new Error('Expected argument of type InstancePingPkg');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_InstancePingPkg(buffer_arg) {
    return management_Management_pb.InstancePingPkg.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_InstanceProperties(arg) {
    if (!(arg instanceof management_Management_pb.InstanceProperties)) {
        throw new Error('Expected argument of type InstanceProperties');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_InstanceProperties(buffer_arg) {
    return management_Management_pb.InstanceProperties.deserializeBinary(new Uint8Array(buffer_arg));
}
var ManagementServiceService = exports.ManagementServiceService = {
    reportInstanceProperties: {
        path: '/ManagementService/reportInstanceProperties',
        requestStream: false,
        responseStream: false,
        requestType: management_Management_pb.InstanceProperties,
        responseType: common_Common_pb.Commands,
        requestSerialize: serialize_InstanceProperties,
        requestDeserialize: deserialize_InstanceProperties,
        responseSerialize: serialize_Commands,
        responseDeserialize: deserialize_Commands,
    },
    keepAlive: {
        path: '/ManagementService/keepAlive',
        requestStream: false,
        responseStream: false,
        requestType: management_Management_pb.InstancePingPkg,
        responseType: common_Common_pb.Commands,
        requestSerialize: serialize_InstancePingPkg,
        requestDeserialize: deserialize_InstancePingPkg,
        responseSerialize: serialize_Commands,
        responseDeserialize: deserialize_Commands,
    },
};
exports.ManagementServiceClient = grpc.makeGenericClientConstructor(ManagementServiceService);
