// skywalking-backend-js/lib/trace/context/DummyContext.js
"use strict";
/*!
 *
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var DummySpan_1 = tslib_1.__importDefault(require("../../trace/span/DummySpan"));
var Segment_1 = tslib_1.__importDefault(require("../../trace/context/Segment"));
var Tracing_pb_1 = require("../../proto/language-agent/Tracing_pb");
var DummyContext = /** @class */ (function () {
    function DummyContext() {
        this.span = new DummySpan_1.default({
            context: this,
            operation: '',
            type: Tracing_pb_1.SpanType.LOCAL,
        });
        this.segment = new Segment_1.default();
        this.depth = 0;
    }
    DummyContext.prototype.newEntrySpan = function (operation, carrier) {
        return this.span;
    };
    DummyContext.prototype.newExitSpan = function (operation, peer) {
        return this.span;
    };
    DummyContext.prototype.newLocalSpan = function (operation) {
        return this.span;
    };
    DummyContext.prototype.start = function () {
        this.depth++;
        return this;
    };
    DummyContext.prototype.stop = function () {
        return --this.depth === 0;
    };
    DummyContext.prototype.async = function (span) {
        return;
    };
    DummyContext.prototype.resync = function (span) {
        return;
    };
    DummyContext.prototype.currentSpan = function () {
        throw new Error('DummyContext.currentSpan() should never be called!');
    };
    return DummyContext;
}());
exports.default = DummyContext;
